src/apps/TxBuilder/components/Dashboard.tsx

```diff
@@ -151,7 +155,9 @@ const Dashboard = () => {
       description = method.name + " (";
       for (let i = 0; i < method.inputs.length; i++) {
         const cleanValue = inputCache[i] || "";
-        cleanInputs[i] = cleanValue;
+        cleanInputs[i] =
+          // eslint-disable-next-line
+          cleanValue.charAt(0) === "[" ? eval(cleanValue) : cleanValue;
```

Comment:
Is there a chance to avoid using `eval`? If it's not, maybe adding a comment explaining it will be helpful.


Have you tried with `JSON.parse`?

Reply:
yes, I did and it was not working in some cases. I didn't find another alternative than using eval .

Reply:
what happens if you put ['alert('Gotcha')]

Reply:
It will prompt the alert, of course, it's evaluating JS code.
What's **REALLY** important: It's not recognized as a valid parameter and the TX is not added to the list. ;)

Reply:
why is it not possible to expect the array input as json syntax, so that you can do `JSON.parse(cleanValue)`?

Reply:
This is not a valid JSON `json.parse(['1', '2'])`. 
`Uncaught ReferenceError: json is not defined`

Reply:
@nicosampler 
it has to be JSON instead of json
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

Reply:
you're right, but the result it's the same.

![image](https://user-images.githubusercontent.com/4376509/93765635-97bf3b80-fbeb-11ea-81c9-a7da31e937fc.png)

Reply:
I found a way to make it work without `eval` by escaping the `"` character: `JSON.parse(cleanValue.replace(/"/g, '"'))`

Reply:
@nicosampler this snippet of code replaces " with ", doesn't it?

Reply:
<img width="229" alt="Screenshot 2020-09-21 at 17 04 58" src="https://user-images.githubusercontent.com/16622558/93769958-a2e68b80-fc2c-11ea-8605-459a0dff8609.png">

Reply:
yes! The input value provided by the UI does not scape the `"`. So we have two alternatives, request the user to scape the value like so: \\" or let the user include the raw `"` and do the work ourselves.